src/main/java/com/ibm/crail/storage/blkdev/BlkDevStorageClient.java

```diff
@@ -8,22 +8,87 @@
 import com.ibm.crail.storage.blkdev.client.BlkDevStorageEndpoint;
 
 import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+import java.util.HashMap;
+import java.util.List;
 import org.slf4j.Logger;
 
 public class BlkDevStorageClient implements StorageClient{
     private static final Logger LOG = CrailUtils.getLogger();
+	private HashMap<Long, String> nodeMap;
 
 	public void printConf(Logger logger) {
-		BlkDevStorageConstants.printConf(logger);
+		BlkDevStorageConstants.printClientConf(logger);
 	}
 
+	private long convertIPPortToKey(String ipPort) throws IOException {
+		//The first token is the ip and the second is the port
+		StringTokenizer tokenizer = new StringTokenizer(ipPort, ":");
+		String ip = tokenizer.nextToken();
+		byte[] bytes = InetAddress.getByName(ip).getAddress();
+		int port = Integer.parseInt(tokenizer.nextToken());
+
+		return DataNodeInfo.calcKey(bytes, port);	
+	}
+
+	/*
+	 * Read the comma seperated lists of virtual device paths and their corresponding datanodes
+	 * (IP/ports) from the configuration file, creating a map from the key (IP/port) to virtual
+	 * device path. This is map is then consulted to identify the virtual device to when an Endpoint
+	 * is created.
+	 */
+	private void createHT() throws IOException {
+		nodeMap  = new HashMap<Long, String>();
+		
+		StringTokenizer tokenizer = new StringTokenizer(BlkDevStorageConstants.DATA_PATH, ",");

```

Comment:
I would do all the parsing in Constants.java and verify same length, if path exists etc in the constants's verify method.

Reply:
Good point.